docs/devel/automation.md

```diff
@@ -0,0 +1,125 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/devel/automation.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Kubernetes Development Automation
+
+## Overview
+
+Kubernetes uses a variety of automated tools in an attempt to relieve developers of repeptitive, low
+brain power work.  This document attempts to describe these processes.
+
+
+## Submit Queue
+
+In an effort to
+   * reduce load on core developers
+   * maintain e2e stability
+   * load test githubs label feature
+
+We have added an automated [submit-queue](https://github.com/kubernetes/contrib/tree/master/submit-queue)
+for kubernetes.
+
+The submit-queue does the following:
+
+```go
+for _, pr := range readyToMergePRs() {
+    if testsAreStable() {
+        mergePR(pr)
+    }
+}
+```
+
+### Ready to merge status
+
+A PR is considered "ready for merging" if it matches the following:
+   * it has the 'lgtm' label, and that 'lgtm' is newer than the latest commit
+   * it has passed the cla pre-submit and has the 'cla:yes' label
+   * it has passed the travis and shippable pre-submit tests
+   * one (or both) of
+      * It's author is in kubernetes/contrib/submit-queue/whitelist.txt
+      * the PR has the 'ok-to-merge' label
+   * One (or both of)
+      * it has passed the Jenkins e2e test
+      * it has the 'e2e-not-required' label
+
+
+### Merge process
+
+Merges _only_ occur when the 'critical builds' (Jenkins e2e for gce, gke, scalability, upgrade) are passing.
+We're open to including more builds here, let us know...
+
+Merges are serialized, so only a single PR is merged at a time, to ensure against races.
+
+If the PR has the 'e2e-not-required' label, it is simply merged.
+If the PR does not have this label, e2e tests are re-run, if these new tests pass, the PR is merged.
```

Comment:
documented (and agree that it's kind of a bug, but catching flaky tests in the submit-queue requires more infrastructure...)